File: Recursão - Python/fatorial.py
def fatorial(n: int) -> int:
  if n == 0:
    return 1
  if n <= 2:
    return n
  
  indice = 1;
  numero = 1;
  
  while (indice <= n):
    numero = numero * indice
    indice += 1
  return numero


numero = input("")
numero = int(numero)

print(fatorial(numero))

File: Recursão - C/fatorial.c
int fatorial(int numero){
  // ponto de parada
  if(numero == 0)
    return 1;
  if(numero <= 2)
    return numero;
  // recursividade
  int n = 1;
  for(int i = 1; i <= numero; i++)
      n  = n * i;
  return n;
}